manager2/src/app/user/user.component.html

```diff
@@ -402,9 +402,9 @@ <h4>SSH keys</h4>
                     <button class="btn btn-default" (click)="ssh_new_key()">Generate new key</button>
                     <hr/>
                     <h4 style="font-weight: normal;">Download</h4>
-                    <a [routerLink]="['/ssh', {id: user.uid}, 'private']" target="_blank"><button class="btn btn-default">private rsa key</button></a>
-                    <a [routerLink]="['/ssh', {id: user.uid}, 'putty']" target="_blank"><button class="btn btn-default">putty private key</button></a>
-                    <a [routerLink]="['/ssh/', {id: user.uid}, 'public']" target="_blank"><button class="btn btn-default">public key</button></a>
+                    <a [href]="[get_ssh_key(user.uid, 'private')]" target="_blank"><button class="btn btn-default">private rsa key</button></a>
+                    <a [href]="[get_ssh_key(user.uid, 'putty')]" target="_blank"><button class="btn btn-default">putty private key</button></a>
+                    <a [href]="[get_ssh_key(user.uid, 'public')]" target="_blank"><button class="btn btn-default">public key</button></a>
                 </div>
```

Comment:
Did you test it?
Why redirecting to an other page instead of current routing? Redirecting to backend also prevent jwt usage.

Reply:
it is not a redirection, only the link string created from the service.

Reply:
A href is a rediection to backebd, not a route in frontend.
On href click, you are redirected to a new page. Routing is different, it loads the angular component.
It indeed seems there was no ssh route in manager2. I need to check what was done in manager 1. Seems something git wrong in port to new angular version.

Reply:
Oh ok :)  I can add a route in the frontend if needed, please keep me inform